File: Dataset/Leetcode/train/101/227.java
 class Solution {
    public boolean XXX(TreeNode root) {
        boolean f = fun(root.left, root.right);
        return f;
    }

    public boolean fun(TreeNode leftNode, TreeNode rightNode){
        if(leftNode == null && rightNode == null){
            return true;
        }
        if(leftNode == null || rightNode == null || leftNode.val != rightNode.val){
            return false;
        }
        return fun(leftNode.left, rightNode.right) && fun(rightNode.left, leftNode.right);    
    }
}

File: Dataset/Leetcode/test/104/227.cpp
class Solution {
public:
    int XXX(TreeNode* root) {
        int ret = 0;
        if (!root) return ret;

        queue<TreeNode*> que;
        que.push(root);
        while(!que.empty()) {
            int levelSize = que.size();
            for (int i = 0; i < levelSize; ++i) {
                auto u = que.front();
                que.pop();
                if (u->left) que.push(u->left);
                if (u->right) que.push(u->right);
            }
            ret ++;
        }
        return ret;
    }
};

